packages/at_secondary_server/lib/src/notification/stats_notification_service.dart

```diff
@@ -43,30 +45,53 @@ class StatsNotificationService {
   InboundConnectionPool inboundConnectionPool =
       InboundConnectionPool.getInstance();
 
+  static final Duration zeroDuration = Duration(microseconds: 0);
+
   // Counter for number of active monitor connections. Used for logging purpose.
   int numOfMonitorConn = 0;
 
   Notification notification = Notification.empty();
 
+  @visibleForTesting
+  /// Set to true while the job is being scheduled, false once it has been scheduled.
+  bool scheduling = false;
```

Comment:
Could this be made less confusing?

As a native English speaker I can tell the difference between scheduling and scheduled, but once the code below starts using them for conditionals it can get murky what's going on.

I think what we have here is a state machine, and so it might be preferable to have a single variable that holds the state of the machine (and then maybe a switch statement to act on the state).

Reply:
Good suggestion - thanks - done

Reply:
I'm not seeing a new commit pushed yet

Reply:
You're too keen :-) I didn't say I'd pushed it. I will re-request your review once the branch has been pushed and the checks have passed